src/tests/Microsoft.PowerFx.Performance.Tests/PvaPerformance.cs

```diff
@@ -0,0 +1,184 @@
+{
+    [MemoryDiagnoser]
+    [NativeMemoryProfiler]
+    [EtwProfiler] // https://benchmarkdotnet.org/articles/features/etwprofiler.html
+    [CsvExporter] // https://benchmarkdotnet.org/articles/configs/exporters.html
+    [MinColumn]
+    [Q1Column]
+    [MeanColumn]
+    [MedianColumn]
+    [Q3Column]
+    [MaxColumn]
+    [SimpleJob(runtimeMoniker: RuntimeMoniker.NetCoreApp31)]
+    public class PvaPerformance
+    {        
+        private ParserOptions _parserOptions;
+        private RecalcEngine _engine;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _parserOptions = new ParserOptions() { AllowsSideEffects = true, Culture = new CultureInfo("en-US") };
+
+            PowerFxConfig powerFxConfig = new PowerFxConfig(new CultureInfo("en-US"), Features.All);
+
+            for (int i = 0; i < 10000; i++)
+            {
+                TestOptionSetDisplayNameProvider nameProvider = new TestOptionSetDisplayNameProvider(Enumerable.Range(0, 10).Select(j => new TestOption($"Logical{j}", $"Display{j}")).ToImmutableArray());
+                powerFxConfig.AddOptionSet(new OptionSet($"OptionSet{i:0000}", nameProvider));
+            }
+
+            _engine = new RecalcEngine(powerFxConfig);
+        }
+
+        [Benchmark]
+        public RecalcEngine PvaRecalcEngineConstructorWith10KOptionSets()
+        {
+            PowerFxConfig powerFxConfig = new PowerFxConfig(new CultureInfo("en-US"), Features.All);
+
+            for (int i = 0; i < 10000; i++)
+            {
+                TestOptionSetDisplayNameProvider nameProvider = new TestOptionSetDisplayNameProvider(Enumerable.Range(0, 10).Select(j => new TestOption($"Logical{j}", $"Display{j}")).ToImmutableArray());
+                powerFxConfig.AddOptionSet(new OptionSet($"OptionSet{i:0000}", nameProvider));
+            }
+
+            RecalcEngine engine = new RecalcEngine(powerFxConfig);           
+            return engine;
+        }
+
+        [Benchmark]
+        public ParseResult PvaRecalcEngineParse()
+        {
+            var expression = GetRandomExpression();
+            ParseResult parseResult = _engine.Parse(expression, _parserOptions);
+
+            if (!parseResult.IsSuccess)
+            {
+                throw new Exception($"{expression}\r\n{string.Join("\r\n", parseResult.Errors.Select(ee => $"{ee.Message}"))}");
+            }
+
+            return parseResult;
+        }
+
+        [Benchmark]
+        public CheckResult PvaRecalcEngineCheck()
+        {
+            var expression = GetRandomExpression();
+            CheckResult checkResult = _engine.Check(expression, _parserOptions);
+
+            if (!checkResult.IsSuccess)
+            {
+                throw new Exception($"{expression}\r\n{string.Join("\r\n", checkResult.Errors.Select(ee => $"{ee.Message}"))}");
+            }
+
+            return checkResult;
+        }
+
+        private static string GetRandomExpression()
```

Comment:
Tbh, I think that this method will take more time than Check itself that we want to benchmark.
I think you can just hardcode a couple expressions